### PR TITLE
Add Auth options to reporter 

### PR DESCRIPTION
**Pull request by riderx**

### CODE REVIEW

**Comment by archlichking** [comment]
Does this PR mean we cannot report to admiral2 without an`auth`?

**Comment by riderx** [comment]
if you don't have auth nothing will change for you.
I made some change to make it clear

**Comment by archlichking** [comment]
LGTM

**Comment by riderx** [comment]
So someone can merge ?

File: src/reporter.js [hunk 1]
```diff
@@ -11,16 +11,33 @@ var ADMIRAL_PROJECT = process.env.ADMIRAL_PROJECT;
 var ADMIRAL_PHASE = process.env.ADMIRAL_PHASE;
 
 // Optional
-var ADMIRAL_RUN = process.env.ADMIRAL_RUN_ID;
+var ADMIRAL_RUN = process.env.ADMIRAL_RUN_ID.replace(%/%g, "-");
```
Comment (archlichking) [inline src/reporter.js]: we have concern on this `.replace(%/%g, "-")`. what's the purpose of replacing string here?

**Comment by archlichking** [comment]
we have concerns on replacing current global env var like ADMIRAL_RUN and encodeURIComponent(ADMIRAL_PROJECT), it might introduce regression in our current scenarios. Any reason why do we need these replacement?

**Review by archlichking — COMMENTED** [review]
we have concerns on replacing current global env var like ADMIRAL_RUN and encodeURIComponent(ADMIRAL_PROJECT), it might introduce regression in our current scenarios. Any reason why do we need these replacement?

**Comment by riderx** [comment]
sorry it was a mistake from my side, the replace should be delete.
 only in the url the admirals var should be uri encode

**Comment by riderx** [comment]
I just fix it can you try it now ?

**Comment by archlichking** [comment]
👍

**Pull request merged**